http-parser/src/http_parser.mli

```diff
@@ -0,0 +1,24 @@
+(** Attempts to parse a buffer into a HTTP request. If successful, it returns
+    the parsed request and an offset value that indicates the starting point of
+    unconsumed content left in the buffer. *)
+
+type error = Partial | Msg of string
+type chunk_kind = Start_chunk | Continue_chunk of int
+
+type chunk_parser_result =
+  | Chunk_complete of string
+  | Done
+  | Partial_chunk of string * int
+
+val parse_request :
+  ?pos:int -> ?len:int -> string -> (Http.Request.t * int, error) result
+
+val parse_chunk_length :
+  ?pos:int -> ?len:int -> string -> (int * int, error) result
+
+val parse_chunk :
+  ?pos:int ->
+  ?len:int ->
+  string ->
+  chunk_kind ->
+  (chunk_parser_result * int, error) result
```

Comment:
My intention was to not have to use `parse_chunk_length` manually, and instead go via the `parse_chunk` which helps orchestrate threading through the chunk body state

Reply:
I'll also note that I don't make use of any of the chunked parsing code in this PR and I'm limiting this diff to just the request (upto the headers) parsing for now